modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+
+		if ( $calypso ) {
+			remove_menu_page( 'plugins.php' );
+
+			if ( $this->is_api_request ) {
+				remove_submenu_page( 'plugins.php', 'plugins.php' );
+			}
+
+			$count = '';
+			if ( ! is_multisite() && current_user_can( 'update_plugins' ) ) {
+				$update_data = wp_get_update_data();
+				$count       = sprintf(
+					'<span class="update-plugins count-%s"><span class="plugin-count">%s</span></span>',
+					$update_data['counts']['plugins'],
+					number_format_i18n( $update_data['counts']['plugins'] )
+				);
+			}
+
+			/* translators: %s: Number of pending plugin updates. */
+			add_menu_page( esc_attr__( 'Plugins', 'jetpack' ), sprintf( __( 'Plugins %s', 'jetpack' ), $count ), 'activate_plugins', 'https://wordpress.com/plugins/' . $domain, null, 'dashicons-admin-plugins', 65 );
+			$this->migrate_submenus( 'plugins.php', 'https://wordpress.com/plugins/' . $domain );
+		}
+	}
+
+	/**
+	 * Adds Users menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_users_menu( $domain, $calypso = true ) {
+		$users_slug   = $calypso ? 'https://wordpress.com/people/team/' . $domain : 'users.php';
+		$add_new_slug = 'https://wordpress.com/people/new/' . $domain;
+		$profile_slug = $calypso ? 'https://wordpress.com/me' : 'grofiles-editor';
+		$account_slug = $calypso ? 'https://wordpress.com/me/account' : 'grofiles-user-settings';
+
+		if ( current_user_can( 'list_users' ) ) {
+			remove_menu_page( 'users.php' );
+			remove_submenu_page( 'users.php', 'users.php' );
+			remove_submenu_page( 'users.php', 'user-new.php' );
+			remove_submenu_page( 'users.php', 'profile.php' );
+			remove_submenu_page( 'users.php', 'grofiles-editor' );
+			remove_submenu_page( 'users.php', 'grofiles-user-settings' );
+
+			add_menu_page( esc_attr__( 'Users', 'jetpack' ), __( 'Users', 'jetpack' ), 'list_users', $users_slug, null, 'dashicons-admin-users', 70 );
+			add_submenu_page( $users_slug, esc_attr__( 'All People', 'jetpack' ), __( 'All People', 'jetpack' ), 'list_users', $users_slug, null, 5 );
+			add_submenu_page( $users_slug, esc_attr__( 'Add New', 'jetpack' ), __( 'Add New', 'jetpack' ), 'promote_users', $add_new_slug, null, 10 );
+			add_submenu_page( $users_slug, esc_attr__( 'My Profile', 'jetpack' ), __( 'My Profile', 'jetpack' ), 'read', $profile_slug, null, 15 );
+			add_submenu_page( $users_slug, esc_attr__( 'Account Settings', 'jetpack' ), __( 'Account Settings', 'jetpack' ), 'read', $account_slug, null, 20 );
+			$this->migrate_submenus( 'users.php', $users_slug );
+		} else {
+			remove_menu_page( 'profile.php' );
+			remove_submenu_page( 'profile.php', 'grofiles-editor' );
+			remove_submenu_page( 'profile.php', 'grofiles-user-settings' );
+
+			add_menu_page( esc_attr__( 'My Profile', 'jetpack' ), __( 'My Profile', 'jetpack' ), 'read', $profile_slug, null, 'dashicons-admin-users', 70 );
```

Comment:
Could this be problematic with plugins that add sections to the wp-admin profile pages, such as Woo or AMP? How do we handle those scenarios?

Reply:
Yes, it would. I suppose that would be another instance where the Calypso <=> wp-admin link toggle will help. 
/cc @davemart-in

Reply:
e46cbe5